src/v/cluster/archival/archival_policy.cc

```diff
@@ -320,6 +320,25 @@ static ss::future<std::optional<std::error_code>> get_file_range(
             co_return seek_result.error();
```

Comment:
> It's unclear exactly why there are missing batches
...
> But the fix is simple enough

can you help me understand this? if we don't know why something is happening, how can we say its fixed (or perhaps this is in reference to something else)?

Reply:
@andrwng would have more context here- but I believe he is referring to the fact that within this segment of code, which deals with non-compacted segment uploads, we run into this corner case which can be thought of as the result of missing offset ranges, (described in greater detail in the comment [here](https://github.com/redpanda-data/redpanda/pull/23937/files/8b616df65bfbddb5dc526bb1e9313ee8d125e81f#diff-a9ffe9fea0d5216cd88d24d01ac31886b82e6442c9f92bf421f74d67bf591bb5R329-R339)). For this reason, it is unclear exactly why these offset ranges are missing, if not due to compaction.

I believe it was postulated in a slack thread that this case could also be the result of enabling compaction before enabling tiered storage.

Reply:
thanks